src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -62,12 +58,16 @@ pub struct GenericExtremeState<S: StateStore> {
     state_table_order_types: Vec<OrderType>,
 
     /// Number of items in the state including those not in top n cache but in state store.
-    total_count: usize, // TODO(yuchao): is this really needed?
+    total_count: usize,
 
     /// Cache for the top N elements in the state. Note that the cache
     /// won't store group_key so the column indices should be offseted
     /// by group_key.len(), which is handled by `state_row_to_cache_row`.
     cache: Cache<Datum>,
+
+    /// Whether the cache is synced to state table, i.e. elements in the
+    /// cache are the most smallest ones in the state table.
+    cache_synced: bool,
```

Comment:
May add more detailed docs for this.

The `cache_synced` is true, only if:
- the cache is empty and `total_count` is 0, or
- the cache is not empty and elements in the cache are the most top ones in the state table.